src/ha-utils/ha/common/utils/apiutils/RESTUtil.go

```diff
@@ -0,0 +1,126 @@
+package apiutils
+
+import (
+	"bytes"
+	"encoding/json"
+	"ha-utils/ha/common/constants"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+func InvokeRESTAPI(httpMethod string, apiURL string, reqHeader map[string]string, requestBody interface{}) (string, string, bool) {
+
+	var httpClient *http.Client = &http.Client{}
+	var request *http.Request
+	var response *http.Response
+	var responseStatus string
+	var err error
+	var responseStatusSuccessful bool
+
+	log.Println("URL being hit is ", httpMethod, " ", apiURL)
+
+	if requestBody == nil {
+		request, err = http.NewRequest(httpMethod, apiURL, nil)
+	} else {
+		jsonValue, _ := json.Marshal(requestBody)
+		request, err = http.NewRequest(httpMethod, apiURL, bytes.NewBuffer(jsonValue))
+	}
+	if err != nil {
+		log.Println("Error occured while creating new request")
+		return "Error occured while creating new HTTP Request", "", false
+	}
+
+	// TODO: set appropriate content type (may be application/json) if not set
+	// Include all required headers passed by the caller
+	for key, value := range reqHeader {
+		request.Header.Set(key, value)
+	}
+
+	// All required params have been set - call the api.
+	response, err = httpClient.Do(request)
+	if err != nil {
+		log.Println("Error occured while caling the api: ", apiURL, "Error is ", err.Error())
+		return err.Error(), "", false
+	}
+
+	responseStatus = strings.TrimSpace(strings.ToUpper(response.Status))
+	log.Println("HTTP Response Status Code: ", responseStatus)
```

Comment:
having a context here helps - for ex., request URL. Else you might not know this HTTP response is for which call.

Reply:
Added `apiURL` to this log statement.